icu4c/source/common/rbbi.cpp

```diff
@@ -89,9 +90,37 @@ RuleBasedBreakIterator::RuleBasedBreakIterator(RBBIDataHeader* data, UErrorCode
 //
 //-------------------------------------------------------------------------------
 RuleBasedBreakIterator::RuleBasedBreakIterator(UDataMemory* udm, UBool isPhraseBreaking,
+                                               const char* dxs,
         UErrorCode &status) : RuleBasedBreakIterator(udm, status)
 {
     fIsPhraseBreaking = isPhraseBreaking;
+    if (U_FAILURE(status)) {
+        return;
+    }
+    if (dxs != nullptr) {
+        size_t length = uprv_strlen(dxs);
+        // The value should be a list of 4 letter script codes joined by '-'.
+        if (length % 5 != 4) {
+            status = U_ILLEGAL_ARGUMENT_ERROR;
+            return;
+        }
+        size_t items = 1 + length / 5;
+        // Change from "thai" to "[:thai:]" or
+        // "thai-arab" to "[[:thai:][:arab:]]"
```

Comment:
As @macchiati  alluded to elsewhere, we should take into account the Script_Extensions property, not just the Script property (I _think_ just changing that to `[:scx=thai:]` etc. will do the right thing in ICU; @markusicu would know). I am not sure if we need to do some special-casing of Inherited; maybe that works out alright since they are all lb=CM, lb=GL, or lb=ZWJ anyway.

Reply:
(I should come up with a test case that exercises that.)